datalad/support/network.py

```diff
@@ -706,7 +734,22 @@ def localpath(self):
         if not (hostname in (None, '', 'localhost', '::1')
                 or hostname.startswith('127.')):
             raise ValueError("file:// URL does not point to 'localhost'")
-        return self.path
+
+        # RFC1738 and RFC3986 both forbid unescaped backslash characters in
+        # URLs, and therefore also in the path-component of file:-URLs. We
+        # assume here that any backslash present in a file-URL is a relict of a
+        # verbatim copy of a Windows-style path.
+        unified_path = self.path.replace('\\', '/')
+        return url2pathname(unified_path)
+        print("---: self.path:", self.path, file=sys.stderr)
+        print("|1 : unified_path:", unified_path, file=sys.stderr)
+        print("|2 : url2pathname(unified_path):", url2pathname(unified_path), file=sys.stderr)
+        print("|3 : pathname2url(url2pathname(unified_path)):", pathname2url(url2pathname(unified_path)), file=sys.stderr)
+        print("|r : url2pathname(unified_path):", url2pathname(unified_path), file=sys.stderr)
+        pathname = url2pathname(unified_path)
+        if on_windows:
+            return pathname.replace("\\", "/")
```

Comment:
This replace confuses me. Shouldn't it be the other way around, returning the backslashes on windows? Not sure what I am missing here.

Reply:
Sorry, that code is not executed. There is however a similar call in line 747 with an explanation in the leading comment

Reply:
Ah, okay. There the replace is happening in the URL, not the returned path like this line here suggested.

All good then!